app/src/main/java/com/example/caregiver/EditTask.java

```diff
@@ -0,0 +1,167 @@
+package com.example.caregiver;
+
+import android.content.Intent;
+import android.content.SharedPreferences;
+import android.os.Bundle;
+import android.preference.PreferenceManager;
+import android.util.Log;
+import android.view.View;
+import android.widget.ArrayAdapter;
+import android.widget.EditText;
+import android.widget.Spinner;
+import android.widget.TextView;
+
+import androidx.appcompat.app.AppCompatActivity;
+
+import com.google.firebase.database.DatabaseReference;
+import com.google.firebase.database.FirebaseDatabase;
+
+import java.util.HashMap;
+import java.util.Map;
+
+public class EditTask extends AppCompatActivity {
+
+    // fields in layout
+    private EditText taskNameField;
+    private EditText taskNotesField;
+    private EditText caregiveeField;
+    private Spinner roomSpinner;
+
+    private TaskFragment.Task currTask; // current task being edited
+    final DatabaseReference database = FirebaseDatabase.getInstance().getReference();
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_edit_task);
+
+        // retrieve info from TaskFragment page intent
+        Intent intent = getIntent();
+        currTask = intent.getParcelableExtra("currTask");
+        String[] caregiveeRooms = intent.getStringArrayExtra("rooms");
+        String caregiveeName = intent.getStringExtra("caregiveeName");
+
+        /* TODO better error handling */
+        // if task is not found, do not display
+        if(currTask == null) return;
+
+        // get all field/spinners
+        taskNameField = findViewById(R.id.taskName);
+        taskNotesField = findViewById(R.id.taskNotes);
+        roomSpinner = findViewById(R.id.taskRoom);
+        caregiveeField = findViewById(R.id.taskCaregivee);
+
+        // populate all fields with task info & create spinner
+        taskNameField.setText(currTask.taskName, TextView.BufferType.EDITABLE);
+        taskNotesField.setText(currTask.taskNote, TextView.BufferType.EDITABLE);
+        caregiveeField.setText(caregiveeName, TextView.BufferType.NORMAL);
+        createSpinner(caregiveeRooms, currTask.room);
+
+        Log.i("INFO", "EditTask:onCreate displayed all task details successfully.");
```

Comment:
Hmm, I think we should remove info logs in code after debugging since it make code cleaner? We should definitely left error logs though so it helps when app crashed.

Reply:
Sure thing, I'll put some error logs in instead